src/v/cluster/controller_backend.cc

```diff
@@ -885,522 +690,491 @@ void controller_backend::housekeeping() {
 
         if (partition) {
-            if (partition->get_revision_id() >= target_rev) {
-                co_return errc::success;
+            rs.set_cur_operation(
+              last_cmd_revision, partition_operation_type::reset, assignment);
+            co_await shutdown_partition(
+              std::move(partition), last_cmd_revision);
+        } else if (claim_it != _ntp_claims.end()) {
+            vlog(
+              clusterlog.trace,
+              "[{}] releasing partition claim {}",
+              ntp,
+              claim_it->second);
+
+            if (claim_it->second.hosting) {
+                claim_it->second.state = partition_claim_state::released;
+            } else {
+                _ntp_claims.erase(claim_it);
+                claim_it = _ntp_claims.end();
             }
+        }
+
+        co_return ss::stop_iteration::yes;
+    }
 
-            // The partition is of older revision, we need to re-create it
-            partition = nullptr;
-            co_await delete_partition(
-              ntp, target_rev, partition_removal_mode::local_only);
-            co_return co_await create_partition(
-              ntp, target_assignment.group, target_rev, cmd_revision, {});
+    // After this point the partition object is expected to exist on current
+    // shard, create it if needed.
+
+    if (!partition) {
+        rs.set_cur_operation(
+          last_cmd_revision, partition_operation_type::reset, assignment);
+
+        replicas_t initial_replicas;
+        if (orig_shard_on_cur_node) {
+            initial_replicas = *orig_replicas;
+        } else if (
+          updated_replicas
+          && update_it->second.get_state()
+               == reconfiguration_state::force_update) {
+            // For force-update new replica starts with the updated
+            // configuration right away (joint consensus is not used)
+            initial_replicas = *updated_replicas;
+        } else {
+            // Configuration will be replicate to the new replica
+            initial_replicas = {};
         }
 
-        bool is_cross_core_target = contains_node(prev_replicas, _self)
-                                    && !has_local_replicas(
-                                      _self, prev_replicas);
-        if (is_cross_core_target) {
-            auto previous_shard = get_target_shard(_self, prev_replicas);
-            co_return co_await create_partition_from_remote_shard(
-              ntp, cmd_revision, *previous_shard, target_assignment);
+        auto ec = co_await create_partition(
+          ntp,
+          assignment.group,
+          expected_log_rev.value(),
+          last_cmd_revision,
+          std::move(initial_replicas));
+        if (ec) {
+            co_return ec;
         }
 
-        co_return co_await create_partition(
-          ntp, target_assignment.group, target_rev, cmd_revision, {});
-    } else {
-        // we need to remove the partition
+        rs.mark_properties_reconciled(tt_rev);
```

Comment:
to make sure i undestand it correctly, we mark properties as reconcilied as the new partition instance is going to be created with properties that are already up to date ?

Reply:
Exactly, we just created a partition with properties queried from topic table at revision `tt_rev`, so all previous properties update requests with smaller or equal revisions are automatically fulfilled.